# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CacheIR.cpp
# Commit: ec930acbd073
# Full Hash: ec930acbd073678eda29f60e7e6b0d89988a02ca
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-01-15 03:50:53
# Description:
#   Bug 1685925 - Only tryAttachTypedArrayLength for specialized stubs. r=iain
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101344
# ==============================================================================

diff -r 434b1d0b2971 -r ec930acbd073 js/src/jit/CacheIR.cpp
--- a/js/src/jit/CacheIR.cpp	Thu Jan 14 16:22:23 2021 +0000
+++ b/js/src/jit/CacheIR.cpp	Thu Jan 14 16:25:24 2021 +0000
@@ -1721,6 +1721,10 @@
     return AttachDecision::NoAction;
   }
 
+  if (mode_ != ICState::Mode::Specialized) {
+    return AttachDecision::NoAction;
+  }
+
   // Receiver should be the object.
   if (isSuper()) {
     return AttachDecision::NoAction;
